File: ProjetoSaúde/processar-links.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Script para extrair coordenadas dos links do Google Maps
e atualizar mapa.js automaticamente
"""

import re
import os
from pathlib import Path

def processar_links():
    # Caminhos
    projeto_dir = Path(__file__).parent
    links_file = projeto_dir / "links-maps" / "links.txt"
    mapa_file = projeto_dir / "js" / "mapa.js"

    if not links_file.exists():
        print(f"❌ Arquivo não encontrado: {links_file}")
        return

    if not mapa_file.exists():
        print(f"❌ Arquivo não encontrado: {mapa_file}")
        return

    # Ler links
    with open(links_file, "r", encoding="utf-8") as f:
        linhas = f.readlines()

    # Ler mapa.js
    with open(mapa_file, "r", encoding="utf-8") as f:
        mapa_content = f.read()

    print("🗺️  Extraindo coordenadas dos links do Google Maps...\n")
    print("=" * 80)

    updated = 0
    coordenadas = []

    # Processar cada linha
    for linha in linhas:
        linha = linha.strip()
        if not linha or "|" not in linha:
            continue

        partes = linha.split("|", 1)
        nome = partes[0].strip()
        url = partes[1].strip() if len(partes) > 1 else ""

        if not url or not url.startswith("http"):
            continue

        # Extrair coordenadas da URL usando regex
        # Formato: /@LAT,LNG,
        match = re.search(r"/@(-?\d+\.\d+),(-?\d+\.\d+)", url)

        if match:
            lat = float(match.group(1))
            lng = float(match.group(2))

            print(f"✅ {nome}")
            print(f"   📍 {lat}, {lng}\n")

            coordenadas.append({
                "nome": nome,
                "lat": lat,
                "lng": lng
            })
        else:
            print(f"❌ {nome}: Coordenadas não encontradas\n")

    print("=" * 80)
    print("\n🔄 Atualizando mapa.js...\n")

    # Atualizar mapa.js para cada unidade
    for coord in coordenadas:
        nome_escaped = re.escape(coord["nome"])

        # Padrão para atualizar lat
        lat_pattern = rf'(nome: "{nome_escaped}"[^}}]*?lat: )-?\d+\.?\d*,'
        lat_replacement = rf'$1{coord["lat"]},'
        before = mapa_content
        mapa_content = re.sub(lat_pattern, lat_replacement, mapa_content)

        # Padrão para atualizar lng
        lng_pattern = rf'(nome: "{nome_escaped}"[^}}]*?lng: )-?\d+\.?\d*,'
        lng_replacement = rf'$1{coord["lng"]},'
        mapa_content = re.sub(lng_pattern, lng_replacement, mapa_content)

        if mapa_content != before:
            print(f"✅ {coord['nome']}: {coord['lat']}, {coord['lng']}")
            updated += 1

    # Salvar arquivo atualizado
    with open(mapa_file, "w", encoding="utf-8") as f:
        f.write(mapa_content)

    print("\n" + "=" * 80)
    print("✅ SUCESSO!")
    print("=" * 80)
    print(f"📍 {updated} unidades atualizadas")
    print(f"📁 Arquivo atualizado: js/mapa.js")
    print("\n🚀 Seu mapa está pronto com coordenadas precisas!")
    print("=" * 80)

if __name__ == "__main__":
    processar_links()

File: ProjetoSaúde/processar-links.js
/**
 * Script para extrair coordenadas dos links do Google Maps
 * e atualizar automaticamente o mapa.js
 * 
 * Execução: node processar-links.js
 */

const fs = require('fs');
const path = require('path');

function procesarLinks() {
    const linksPath = path.join(__dirname, 'links-maps', 'links.txt');
    const mapaPath = path.join(__dirname, 'js', 'mapa.js');

    if (!fs.existsSync(linksPath)) {
        console.error('❌ Arquivo links-maps/links.txt não encontrado!');
        process.exit(1);
    }

    const linksContent = fs.readFileSync(linksPath, 'utf8');
    let mapaContent = fs.readFileSync(mapaPath, 'utf8');

    const linhas = linksContent.trim().split('\n');
    const coordenadas = [];
    let updated = 0;

    console.log('🗺️  Extraindo coordenadas dos links...\n');
    console.log('=' .repeat(80));

    // Processar cada linha
    for (const linha of linhas) {
        if (!linha.trim()) continue;

        const [nome, url] = linha.split('|');
        if (!nome || !url) {
            console.log(`⚠️  Linha inválida (falta | entre nome e URL): ${linha}`);
            continue;
        }

        // Regex para extrair coordenadas da URL do Google Maps
        // Formato: /@LAT,LNG, ou @LAT,LNG,
        const coordRegex = /@(-?\d+\.\d+),(-?\d+\.\d+)/;
        const match = url.match(coordRegex);

        if (match) {
            const lat = parseFloat(match[1]);
            const lng = parseFloat(match[2]);

            console.log(`✅ ${nome.trim()}`);
            console.log(`   📍 ${lat}, ${lng}\n`);

            coordenadas.push({
                nome: nome.trim(),
                lat: lat,
                lng: lng
            });
        } else {
            console.log(`❌ ${nome.trim()}: Coordenadas não encontradas na URL\n`);
        }
    }

    console.log('=' .repeat(80));
    console.log('\n🔄 Atualizando mapa.js...\n');

    // Atualizar o mapa.js
    for (const coord of coordenadas) {
        const escapedNome = coord.nome.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');

        // Padrão para encontrar e atualizar lat
        const latPattern = new RegExp(
            `(nome: "${escapedNome}"[^}]*?lat: )-?\\d+\\.?\\d*,`,
            'g'
        );
        const latReplacement = `$1${coord.lat},`;
        const beforeLat = mapaContent;
        mapaContent = mapaContent.replace(latPattern, latReplacement);

        // Padrão para encontrar e atualizar lng
        const lngPattern = new RegExp(
            `(nome: "${escapedNome}"[^}]*?lng: )-?\\d+\\.?\\d*,`,
            'g'
        );
        const lngReplacement = `$1${coord.lng},`;
        mapaContent = mapaContent.replace(lngPattern, lngReplacement);

        if (mapaContent !== beforeLat) {
            console.log(`✅ ${coord.nome}: ${coord.lat}, ${coord.lng}`);
            updated++;
        }
    }

    // Salvar arquivo atualizado
    fs.writeFileSync(mapaPath, mapaContent);

    console.log('\n' + '=' .repeat(80));
    console.log('✅ SUCESSO!');
    console.log('=' .repeat(80));
    console.log(`📍 ${updated} unidades atualizadas`);
    console.log(`📁 Arquivo atualizado: js/mapa.js`);
    console.log('\n🚀 Seu mapa está pronto com coordenadas precisas!');
    console.log('=' .repeat(80));
}

procesarLinks();
